### ISSUE TITLE
Findings for accessibility staging review for Facility Locator updates

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#10377]
**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

@1Copenut — ready for your VSP review.

## Description

This is the collected findings from the accessibility staging review.

## Point of Contact

**VFS Point of Contact:** Jennifer

<hr>

### Features/enhancements
- Community care pharmacies (in VA's network) is hidden as a facility type for searching but has been feature flagged to return at the end of the month [#9177](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/9177)
- [Facility Detail page] Update to display of hours [#3541](https://github.com/department-of-veterans-affairs/va.gov-team/issues/3541)
- Addition of chatbot link in intro text (this work was done by separate team)
- Addition of pins which correlate facility locations on the map to those within the search results list. [#5620](https://github.com/department-of-veterans-affairs/va.gov-team/issues/5620)
- Sighted veterans can view a pin marking their geographic location as specified in the search parameters so they know their location relevant to search results. [#5501](https://github.com/department-of-veterans-affairs/va.gov-team/issues/5501)
### Resolution of accessibility issues
- Search results lists are presented as ordered list with contextual information for Veterans using screen reader devices. [#5555](https://github.com/department-of-veterans-affairs/va.gov-team/issues/5555)
- Heading changes made to the "Important information about your Community Care appointment" alert to support understanding of page structure. [#7244](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7244)
- Veterans using screen readers are able to navigate away from the search results using a keyboard.
[#7288](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7288)
- Veteran should be able to traverse the options in the dropdown using keyboard only [#7291](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7291)
### Out of scope
- Addition of Facility operating status and notes  (had separate [staging review])(https://github.com/department-of-veterans-affairs/va.gov-team/issues/8099)

## Artifacts - _please provide the following artifacts at least 2 days before scheduled review meeting_
- [x] Access information for the tool: https://www.va.gov/find-locations
   - Facility Locator is unauthenicated and does not need access credentials

**Required accessibility tests**
- [x] Scan with axe browser plugin
- [ ] Create end-to-end tests for React apps (ignore if not applicable)
- [x] Color comparisons
- [ ] Zoom layouts
- [x] Keyboard navigation
- [ ] Screen reader

**Recommended tests**
- [x] Scan with axe-coconut
- [x] Visually scan with WAVE

**Experimental tests**
- [ ] Create keyboard end-to-end tests

## List of known issues

  - **508-defect-1** Focus MUST move to notification of context change when new content is added [#7243](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7243)
  - **508-defect-1** Upon Submit, focus MUST move to the context change notification [#7240](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7240)
  - 508-defect-2 Supplementing the text of the Directions link [#8243](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8243)
  - 508-defect-2 Map is cut off rather than scrolling horizontally. [#6184](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6184)

  **Mapbox-related accessibility issues**
    Mapbox functionality has a number of accessibility defects related to leaflet limitations which will be explored in [#8535](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8535)
  - 508-defect-2 The map is announcing a lot of extra data on focus and each droplet needs a usable name [#515](https://github.com/department-of-veterans-affairs/va.gov-team/issues/515)
  - 508-defect-2 Zoom In / Zoom Out controls MUST have their visible text as part of their accessible name [#6104](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6104)
  - 508-defect-2 Links in map attribution SHOULD open in new tab to aid in error recovery [#6158](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6158)
  - 508-defect-3 Navigating by keyboard between the map and the search results MUST be intuitive with changes communicated [#6897](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6897)
  - 508-defect-4 The Leaflet container SHOULD have role appropriate for interactive content [#6103](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6103)
  - 508-defect-4 CONSIDER changing background of map attribution [#6892](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6892)

### Other bugs
- Short list of health services should not have accordion [#10094](https://github.com/department-of-veterans-affairs/va.gov-team/issues/10094)
- Results summary subhead has been suppressed [#6775](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6775)
- Vet Center "service type" should not be interactive [#5967](https://github.com/department-of-veterans-affairs/va.gov-team/issues/5967)
- Phone numbers on the Facility Detail page need to be linked and many are displayed twice: once as link for auto-dialer and one as routine display for "Main phone number". [#8063](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8063)
- Some cemetery searches may not yield results due to GeoBISL data freshness. This will be resolved by migration to v1. [#9662](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9662)
- [Facility detail page] Accordions bug [#10094](https://github.com/department-of-veterans-affairs/va.gov-team/issues/10094)

**Comment by jenstrickland** [comment]
@mmiddaugh I will be late with my audit. My apologies! There's just so many other things. This is not to say that this isn't important.

**Comment by jenstrickland** [comment]
**Working notes in progress**

- axe and axe-coconut scans of the initial FL page, populated search results, and the facility detail page were all clean, apart from documented known issues above

**Comment by jenstrickland** [comment]
@mmiddaugh Additionally, this article may be of interest for your team:

https://equalentry.com/accessible-maps-on-the-web/

**Comment by 1Copenut** [comment]
@jenstrickland I've reviewed these issues and commented on individual tickets. The only thing that I noticed was the tabbed list or map behavior on mobile device, that didn't appear when I zoomed my layout. This seems like we're doing some type of viewport detection on first load to show or hide the tabs, but not also doing that detection on viewport resize? (Complete guess). I think your tickets have accounted for this discrepancy and look forward to seeing the fixes.

**Comment by jenstrickland** [comment]
@1Copenut As I understand it, that was existing Facility Locator mobile behavior — to provide the map in a tab. In a previous recommendation I suggested incorporating something similar for the full design, so that the map is provided on user action and more search results are visible. I believe there's research indicating for rural areas this isn't a great UX, but my memory may be faulty.

**Comment by mmiddaugh** [comment]
Responding to your [comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/10377#issuecomment-649818895) about phone numbers for community care detail pages...

Thank you for that catch! 
FYI, we will be hiding Community Care detail pages in the next sprint or two so that particular issue will soon be irrelevant.

**Comment by laflannery** [comment]
Closing as there are no open issues remaining in the epic
